### PR TITLE
*: remove call of SchemaTableInfos for CheckPlacementPolicy

### PR DESCRIPTION
**Pull request by joechenrh**
### What problem does this PR solve?

Issue Number: ref #55394

Problem Summary:

### What changed and how does it work?

Replace`SchemaTableInfos` with `ListTablesWithSpecialAttribute` to get table infos.

Currently, `CheckPlacementPolicyNotInUseFromInfoSchema` will fetch table infos from meta, which makes it no different from `CheckPlacementPolicyNotInUseFromMeta`.

https://github.com/pingcap/tidb/blob/87244edabf0ebae9870411cb83d91a9add4a797e/pkg/ddl/placement_policy.go#L347-L362

Additionally, a new function `GetTableInfoWithAttributes` is added to meta module to retrieve tables with special attributes under certain db.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [X] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Test using the following SQLs:

```mysql
mysql> use test
Database changed

mysql> CREATE PLACEMENT POLICY p1 PRIMARY_REGION="us-east-1" REGIONS="us-east-1,us-west-1" FOLLOWERS=4;
Query OK, 0 rows affected (0.02 sec)

mysql> CREATE PLACEMENT POLICY p3 FOLLOWERS=4;
Query OK, 0 rows affected (0.02 sec)

mysql> CREATE TABLE t1 (a INT) PLACEMENT POLICY=p3;
Query OK, 0 rows affected (0.03 sec)

mysql> drop placement policy p1;
Query OK, 0 rows affected (0.05 sec)

mysql> drop placement policy p3;
ERROR 8241 (HY000): Placement policy 'p3' is still in use
```

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/ddl/placement_policy.go [hunk 1]
```diff
@@ -367,17 +367,17 @@ func CheckPlacementPolicyNotInUseFromInfoSchema(is infoschema.InfoSchema, policy
...
 		}
+	}
 
-		tblInfos, err := is.SchemaTableInfos(context.Background(), dbInfo.Name)
-		if err != nil {
-			return errors.Trace(err)
-		}
-		for _, tblInfo := range tblInfos {
+	schemaTables := is.ListTablesWithSpecialAttribute(infoschema.AllPlacementPolicyAttribute)
+	for _, schemaTable := range schemaTables {
```
Comment (CbcWestwolf) [inline pkg/ddl/placement_policy.go:373]: Should we filter with `dbInfo.Name` here?

Comment (joechenrh) [reply pkg/ddl/placement_policy.go:373]: Here just retrieve and check all tables with necessary attributes, so it's seperated from the previous loop.

**Review by CbcWestwolf — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Review by tangenta — APPROVED** [review]

**Comment by fzzf678** [comment]
It seems no unit test added in this pr, please update the Check List in description

**Pull request merged**